### PR TITLE
nix-weather: init at 0.0.3

### PR DESCRIPTION
**Pull request by cafkafk**
## Description of changes

Changelog: https://git.fem.gg/cafkafk/nix-weather/releases/tag/v0.0.3 https://git.fem.gg/cafkafk/nix-weather/releases/tag/v0.0.2

Signed-off-by: Christina Sørensen <christina@cafkafk.com>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by cafkafk** [comment]
@freyacodes I've added you as maintainer and co-author on this, and included some of the changes from #340477 (see https://github.com/NixOS/nixpkgs/compare/639b2f1a5c19cbeb760eaa63b24929c794e816df..5658c35d7fa10816a0d11d283ab197f6b7070a78). Hope that's okay :3

**Review by freyacodes — APPROVED** [review]
New changes look good. It builds and runs just fine :+1:

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 340518` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>nix-weather</li>
    <li>nix-weather.man</li>
  </ul>
</details>

File: pkgs/by-name/ni/nix-weather/package.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs =
+    [
+      openssl
+      installShellFiles
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      libiconv
+      darwin.apple_sdk.frameworks.Security
```
Comment (wegank) [inline pkgs/by-name/ni/nix-weather/package.nix:38]:
```suggestion
      darwin.apple_sdk.frameworks.Security
      darwin.apple_sdk.frameworks.SystemConfiguration
```

Comment (cafkafk) [reply pkgs/by-name/ni/nix-weather/package.nix:38]: Thanks! Addressed by https://github.com/NixOS/nixpkgs/compare/5658c35d7fa10816a0d11d283ab197f6b7070a78..227afc48a21338210db6ff24945f926076a1bff8

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 340518` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nix-weather</li>
    <li>nix-weather.man</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 340518` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nix-weather</li>
    <li>nix-weather.man</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**